src/coreclr/vm/stubmgr.cpp

```diff
@@ -938,10 +938,15 @@ BOOL ThePreStubManager::DoTraceStub(PCODE stubStartAddress, TraceDestination *tr
     // We cannot tell where the stub will end up
     // until after the prestub worker has been run.
     //
+    #if defined(TARGET_ARM64) && defined(__APPLE__)
+        // On ARM64 Mac, we cannot put a breakpoint inside of ThePreStubPatchLabel
+        LOG((LF_CORDB, LL_INFO10000, "TPSM::DoTraceStub: Skipping on M1\n"));
+        return FALSE;
+    #else
+        trace->InitForFramePush(GetEEFuncEntryPoint(ThePreStubPatchLabel));
 
-    trace->InitForFramePush(GetEEFuncEntryPoint(ThePreStubPatchLabel));
-
-    return TRUE;
+        return TRUE;
+    #endif
```

Comment:
This isn't always consistent, but it is helpful to at least mark the `#endif` with a comment of the associated predicate. This line could be:

```suggestion
#endif // defined(TARGET_ARM64) && defined(__APPLE__)
```

An example in the repo

https://github.com/dotnet/runtime/blob/8f74726e77b7ef4c00b90ad852b3917727e11e0c/src/coreclr/vm/dllimport.cpp#L2477